### PR TITLE
no need check is nil, because has checked before

### PR DESCRIPTION
**Pull request by ghost**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/priorities/node_prefer_avoid_pods.go [hunk 1]
```diff
@@ -51,10 +51,8 @@ func CalculateNodePreferAvoidPodsPriorityMap(pod *v1.Pod, meta interface{}, node
 	}
 	for i := range avoids.PreferAvoidPods {
 		avoid := &avoids.PreferAvoidPods[i]
-		if controllerRef != nil {
-			if avoid.PodSignature.PodController.Kind == controllerRef.Kind && avoid.PodSignature.PodController.UID == controllerRef.UID {
-				return schedulerapi.HostPriority{Host: node.Name, Score: 0}, nil
-			}
+		if avoid.PodSignature.PodController.Kind == controllerRef.Kind && avoid.PodSignature.PodController.UID == controllerRef.UID {
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/node_prefer_avoid_pods.go]: The `controllerRef` maybe nil according to previous codes. So this PR is not valid to me :).

Comment (weizhanxu) [inline plugin/pkg/scheduler/algorithm/priorities/node_prefer_avoid_pods.go]: if controllerRef == nil {
		return schedulerapi.HostPriority{Host: node.Name, Score: 10}, nil
	}

	avoids, err := v1helper.GetAvoidPodsFromNodeAnnotations(node.Annotations)
	if err != nil {
		// If we cannot get annotation, assume it's schedulable there.
		return schedulerapi.HostPriority{Host: node.Name, Score: 10}, nil
	}
	for i := range avoids.PreferAvoidPods {
		avoid := &avoids.PreferAvoidPods[i]
		if controllerRef != nil {
			if avoid.PodSignature.PodController.Kind == controllerRef.Kind && avoid.PodSignature.PodController.UID == controllerRef.UID {
				return schedulerapi.HostPriority{Host: node.Name, Score: 0}, nil
			}
		}
	}
	return schedulerapi.HostPriority{Host: node.Name, Score: 10}, nil
}

the controllerRef is a local var, up code is part of the fuction. before my modify code, it has checked nil,  function will be return when nil.

Comment (k82cn) [inline plugin/pkg/scheduler/algorithm/priorities/node_prefer_avoid_pods.go]: got that, LGTM.

**Comment by k82cn** [comment]
btw, please signed CLA.

**Comment by weizhanxu** [comment]
cla test

**Comment by k82cn** [comment]
@k8s-bot ok to test

**Pull request merged**